### ISSUE TITLE
Validate -blocks-storage.bucket-store.ignore-blocks-within when running in monolithic mode

### ISSUE THREAD

**Issue by pracucci** [grafana/mimir#2675]
I'm seeing help requests from several people lowering `-querier.query-store-after` without changing `-blocks-storage.bucket-store.ignore-blocks-within`, causing some queries to fail with "consistency check failed" errors.

We should validate `-blocks-storage.bucket-store.ignore-blocks-within` setting to make sure it has been set to a value lower than `-querier.query-store-after` minus a delta ([in jsonnet we used a 2h delta](https://github.com/grafana/mimir/pull/1915/files#diff-c2844a5133c123cdd52a805fa70f5d2d858bd234ad3af6b8827d355ce10bfa37)).

Unfortunately, `-querier.query-store-after` is set on the querier, while `-blocks-storage.bucket-store.ignore-blocks-within` is set on the store-gateway, so there's no reliable way to validate it when running in microservices mode. However, we could at least validate it when running monolithic mode.

**Comment by Andysimcoe** [comment]
In case anyone is searching the docs and then finds this like I did, this did resolve the issue, I hadn't seen it before and hadn't recently changed the config.

This is now resolved lowering `ignore_blocks_within` to an hour less than `query_store_after`. I have these set in a shared config so I guess validation could also work for that too, albeit it won't catch CLI flags for monolithic mode but for those of us setting it in the config file it may still be a nice catch.

**Comment by pracucci** [comment]
> albeit it won't catch CLI flags for monolithic mode but for those of us setting it in the config file it may still be a nice catch.

Agree. The problem of having this validation check for the non-monolithic deployment mode is that `ignore_blocks_within` and `query_store_after` could be not set on all components and this could make fail the validation check on some components (but would be a false positive even if these configs are correctly set on the right components). This wouldn't be a problem with the monolithic mode because being monolithic it requires all the config.

**Comment by narqo** [comment]
While doesn't cover 100% of cases, I think, a lot of users run Mimir passing a static YAML config (ie. what the Mimir's Helm chart is doing). A validation on the config level — in addition to a validation in Jsonnet — would cover a lot of practical scenarios for µ-services mode.
